packages/core/data-transfer/src/engine/index.ts

```diff
@@ -88,13 +88,19 @@ class TransferEngine<
 
   #metadata: { source?: IMetadata; destination?: IMetadata } = {};
 
+  // Progress of the current stage
   progress: {
+    // metrics on the progress such as size and record count
     data: TransferProgress;
+    // stream that emits events
     stream: PassThrough;
   };
 
   diagnostics: IDiagnosticReporter;
 
+  // Save the currently open stream so that we can access it at any time
+  #currentStream?: Writable;
```

Comment:
We might have errors in the future if we have multiple streams transferring data at the same time.
Might have been worth to use an array or an object instead but I guess it'll work for now